docs/architecture/commitlog/fileformat.rst

```diff
@@ -0,0 +1,63 @@
+ScyllaDB commitlog segment file format
+======================================
+
+**Note:** Commitlog file formats are subject to change between scylla versions. Users should not make assumptions about nor rely on them.
+Commitlog files should *never* be used across ScyllaDB updates. This information is provided mainly for ScyllaDB contributors.
```

Comment:
@elcallio so now I'm curious how we handle upgrades - when we bring down a node to upgrade it, how do we ensure that no commit logs remain? Is there some sort of "drain" (is this the right word?) step to ensure that the memtables are all flushed and the commit logs deleted before the node goes down for an upgrade? Is this necessary step a documented part of the upgrade process?

People reading this document might be curious, like me, how we ensure that commit logs aren't present during upgrades so it might be worth mentioning how this is solved.

Reply:
Honestly, I have no idea which aspects in the docs folder are supposed to be of end-user interest or not. I placed it there in symmetry with sstable file formats, which I felt was the closest thing in similarity. It is most definitely of developer-only interest.

Reply:
But I realize now that there is a non-user-facing md-only similar file format doc in the "dev" folder. Seems things should be there.

Reply:
Regarding upgrade: Any clean shutdown of scylla will remove all data from all commitlogs. That is what the memtable flush:es on shutdown are for. Commitlog files (even hints) should only remain on disk as a result of a non-clean shutfown (i.e. crash).